lib/api/apiUtils/bucket/bucketDeletion.js

```diff
@@ -151,12 +152,23 @@ export function deleteBucket(bucketMD, bucketName, canonicalID, log, cb) {
                     return _deleteMPUbucket(bucketName, log, next);
                 });
         },
+        function destroyBucketKey(next) {
```

Comment:
Are object PUTs allowed on buckets with delete flags?

Apart from that I agree with deleting the key after the bucket is deleted or marked for deletion.